File: OAIP/LR-3-4/2 лаба 3.15.py
def kiosk_sales():
    newspaper_price = 30.0
    magazine_price = 20.0
    total_sales = 0
    total_revenue = 0.0
    while True:
        choice = input('Что вы хотите купить? (G - газета, M - журнал): ').strip().upper()
        if choice == 'G':
            print(f'Цена газеты: {newspaper_price:.2f} руб.')
            payment = float(input('Введите сумму оплаты: '))
            if payment >= newspaper_price:
                total_revenue += newspaper_price
                total_sales += 1
                print(f'Спасибо за покупку! Сдача: {(payment - newspaper_price):.2f} руб.')
            else:
                print('Недостаточно средств для покупки газеты.')
        elif choice == 'M':
            print(f'Цена журнала: {magazine_price:.2f} руб.')
            payment = float(input('Введите сумму оплаты: '))
            if payment >= magazine_price:
                total_revenue += magazine_price
                total_sales += 1
                print(f'Спасибо за покупку! Сдача: {(payment - magazine_price):.2f} руб.')
            else:
                print('Недостаточно средств для покупки журнала.')
        else:
            print('Некорректный выбор. Пожалуйста, выберите G или M.')
        continue_selling = input('Хотите продолжить продажи? (Y - да, N - нет): ').strip().upper()
        if continue_selling == 'N':
            break
    print(f'Общее количество продаж: {total_sales}')
    print(f'Общая выручка: {total_revenue:.2f} руб.')
if __name__ == "__main__":
    kiosk_sales()

File: OAIP/LR-3-4/2 лаба 3.15.pas
program kiosk_sales;

var
  newspaperPrice, magazinePrice: real; 
  totalSales: integer; 
  totalRevenue: real; 
  choice: char; 
  payment: real; 
  continueSelling: char;

begin
  
  newspaperPrice := 30; 
  magazinePrice := 20; 
  totalSales := 0;
  totalRevenue := 0;
  
  
  repeat
    writeln('Что вы хотите купить? (G - газета, M - журнал): ');
    readln(choice);
    
    if choice = 'G' then
    begin
      writeln('Цена газеты: ', newspaperPrice:0:2, ' руб.');
      writeln('Введите сумму оплаты: ');
      readln(payment);
      
      if payment >= newspaperPrice then
      begin
        totalRevenue := totalRevenue + newspaperPrice;
        totalSales := totalSales + 1;
        writeln('Спасибо за покупку! Сдача: ', (payment - newspaperPrice):0:2, ' руб.');
      end
      else
        writeln('Недостаточно средств для покупки газеты.');
      
    end else if choice = 'M' then
    begin
      writeln('Цена журнала: ', magazinePrice:0:2, ' руб.');
      writeln('Введите сумму оплаты: ');
      readln(payment);
      
      if payment >= magazinePrice then
      begin
        totalRevenue := totalRevenue + magazinePrice;
        totalSales := totalSales + 1;
        writeln('Спасибо за покупку! Сдача: ', (payment - magazinePrice):0:2, ' руб.');
      end
      else
        writeln('Недостаточно средств для покупки журнала.');
      
    end else
      writeln('Некорректный выбор. Пожалуйста, выберите G или M.');
    
    writeln('Хотите продолжить продажи? (Y - да, N - нет): ');
    readln(continueSelling);
    
  until continueSelling = 'N';
  
  
  writeln('Общее количество продаж: ', totalSales);
  writeln('Общая выручка: ', totalRevenue:0:2, ' руб.');
  
  readln; 
end.
